### PR TITLE
Fix issues reported by PHPStan 1.10.39

### PR DESCRIPTION
**Pull request by Screenfeed**

### CODE REVIEW

File: include/license.php [hunk 1]
```diff
@@ -84,10 +84,12 @@ public function __construct( $file, $item_name, $version, $author, $api_url = nu
 		$this->author  = $author;
 		$this->api_url = empty( $api_url ) ? $this->api_url : $api_url;
 
-		$licenses = get_option( 'polylang_licenses' );
-		$this->license_key = empty( $licenses[ $this->id ]['key'] ) ? '' : $licenses[ $this->id ]['key'];
-		if ( ! empty( $licenses[ $this->id ]['data'] ) ) {
-			$this->license_data = $licenses[ $this->id ]['data'];
+		$licenses          = (array) get_option( 'polylang_licenses', array() );
```
Comment (Hug0-Drelon) [inline include/license.php:87]: What happen if `get_option( 'polylang_licenses' )` returns an object here?

Comment (Screenfeed) [reply include/license.php:87]: It becomes an array.

Comment (Hug0-Drelon) [reply include/license.php:87]: Mmmmh... I thought casting an object into an array requires more work to get all properties properly. Thanks.

**Review by Hug0-Drelon — APPROVED** [review]

File: include/license.php [hunk 2]
```diff
@@ -84,10 +84,12 @@ public function __construct( $file, $item_name, $version, $author, $api_url = nu
...
+		$license           = isset( $licenses[ $this->id ] ) && is_array( $licenses[ $this->id ] ) ? $licenses[ $this->id ] : array();
+		$this->license_key = ! empty( $license['key'] ) && is_string( $license['key'] ) ? $license['key'] : '';
+
+		if ( ! empty( $license['data'] ) && $license['data'] instanceof stdClass ) {
+			$this->license_data = $license['data'];
```
Comment (Chouby) [inline include/license.php:92]: Wouldn't this work 
```php
$this->license_key = (string) $license['key'];
$this->license_data = (object) $license['data'];
```

Comment (Screenfeed) [reply include/license.php:92]: `key` and `data` may not be set.

Comment (Chouby) [reply include/license.php:92]: Sure. However, I find:
```php
		$this->license_key = ! empty( $license['key'] ) ? (string) $license['key'] : '';
		
		if ( ! empty( $license['data'] ) ) {
			$this->license_data = (object) $license['data'];
		}
```
more elegant than type check. It's however not strictly equivalent to what you wrote.
As you prefer.

File: include/license.php [hunk 3]
```diff
@@ -224,7 +226,7 @@ private function api_request( $request ) {
 			$licenses[ $this->id ] = array( 'key' => $this->license_key );
 			$data = json_decode( wp_remote_retrieve_body( $response ) );
 
-			if ( 'deactivated' !== $data->license ) {
+			if ( $data instanceof stdClass && isset( $data->license ) && 'deactivated' !== $data->license ) {
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply include/license.php]: Done in [934a90f](https://github.com/polylang/polylang/pull/1372/commits/934a90f5c418d12234f01ab85308c576a3e636db).

**Pull request merged**